core/shared/src/main/scala/org/http4s/multipart/PartReceiver.scala

```diff
@@ -0,0 +1,206 @@
+    * passing `strict = false` to the underlying `decode` method.
+    *
+    * This method passes `strict = false` to the `decoder.decode` method.
+    *
+    * @param decoder The delegate EntityDecoder
+    */
+  def decode[F[_], A](implicit decoder: EntityDecoder[F, A]): PartReceiver[F, A] =
+    part => Resource.eval(decoder.decode(part, strict = false).value)
+
+  /** Creates a PartReceiver that delegates to an implicitly-resolved `EntityDecoder` to decode the part body,
+    * passing `strict = true` to the underlying `decode` method.
+    *
+    * @param decoder The delegate EntityDecoder
+    */
+  def decodeStrict[F[_], A](implicit decoder: EntityDecoder[F, A]): PartReceiver[F, A] =
+    part => Resource.eval(decoder.decode(part, strict = true).value)
+
+  /** Creates a PartReceiver that loads the part's body into an in-memory buffer, up until
+    * a specified maximum size, at which point it will instead write the body to a temp file.
+    *
+    * The temp file created by this decoder (if any) will be released with the Resource returned
+    * by this receiver's `receive` method.
+    *
+    * @param maxSizeBeforeFile Maximum number of bytes to be buffered in memory for each part
+    *                          received by the returned receiver. When the in-memory buffer
+    *                          would exceed this limit, the buffer is dumped to a temp file
+    *                          and all subsequent data pulled from the part body will be
+    *                          appended to that file instead of accumulating the memory buffer.
+    * @param chunkSize The chunk size used when reading data back from a temporary file created by this receiver
+    * @return A PartReceiver which dynamically decides whether to buffer the part's body in memory or in a file
+    */
+  def toMixedBuffer[F[_]](
+      maxSizeBeforeFile: Int,
+      chunkSize: Int = 8192,
+  )(implicit
+      files: Files[F],
+      mt: MonadThrow[F],
+      c: Compiler[F, F],
+  ): PartReceiver[F, Stream[F, Byte]] =
+    part => readToBuffer[F](part.body, maxSizeBeforeFile, chunkSize).map(Right(_))
+
+  private def limitPartSize[F[_]](
+      maxPartSizeBytes: Long
+  )(implicit F: ApplicativeError[F, Throwable]): Pipe[F, Byte, Byte] = {
+    def go(s: Stream[F, Byte], accumSize: Long): Pull[F, Byte, Unit] = s.pull.uncons.flatMap {
+      case Some((chunk, tail)) =>
+        val newSize = accumSize + chunk.size
+        if (newSize <= maxPartSizeBytes) Pull.output(chunk) >> go(tail, newSize)
+        else Pull.raiseError[F](InvalidMessageBodyFailure("Part body exceeds maximum length"))
+      case None =>
+        Pull.done
+    }
+
+    go(_, 0L).stream
+  }
+
+  private def readToBuffer[F[_]: Files: MonadThrow](
+      input: Stream[F, Byte],
+      maxSizeBeforeFile: Int,
+      chunkSize: Int,
+  )(implicit c: Compiler[F, F]): Resource[F, Stream[F, Byte]] = {
+    final case class Acc(bytes: Stream[Pure, Byte], bytesSize: Int)
```

Comment:
According to the way how it is used in the code, it seems that the entire `Acc` class can be replaced with just one `Chunk`, which already is aware of its size therefore no additional field is required to pass it around.